### PR TITLE
[master] Maven build - release version property change to 3.1.0

### PR DESCRIPTION
**Pull request by rfelcman**
This is small fix to reflect Maven project 3.1.0-SNAPSHOT changed before.

Signed-off-by: Radek Felcman <radek.felcman@oracle.com>

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -98,7 +98,7 @@
         <!-- TOOL Properties -->
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <!-- PROJECT Properties -->
-        <release.version>3.0.1</release.version>
+        <release.version>3.1.0</release.version>
```
Comment (lukasj) [inline pom.xml:101]: Would it make sense to have this set to ‘${project.version}’ instead?

Comment (rfelcman) [reply pom.xml:101]: Sorry, but I don't think, that is possible as this property is used in nightly build output directory at Eclipse.org and some POMs/modules in `bundles` subdirectory.

Comment (lukasj) [reply pom.xml:101]: I'm afraid we'll keep forgetting to update this property going forward. What about different solution, ie `osgiversion-maven-plugin` can drop defined parts from a version property (see ie [here](https://github.com/eclipse-ee4j/glassfish-ha-api/blob/master/pom.xml#L174)). Would that be useful for this?

Comment (lukasj) [reply pom.xml:101]: ...but we need ie `-RC` suffix, right?

Comment (rfelcman) [reply pom.xml:101]: I strongly agree with "forgetting" comment. I'll try something based on http://www.mojohaus.org/build-helper-maven-plugin/regex-property-mojo.html plugin.
About `-RC` suffix. It depends. In Nightly dir no, in Javadoc headers, in https://github.com/eclipse-ee4j/eclipselink/blob/master/foundation/org.eclipse.persistence.core/src/main/resources/org/eclipse/persistence/version.properties maybe (but there is qualifier).

Comment (lukasj) [reply pom.xml:101]: in the `version.properties`, `version` should be full `${project.version}` instead. Same in javadoc. So it is probably just a folder name, which would be good to figure out ...

Comment (rfelcman) [reply pom.xml]: Please check latest changes.

Comment (lukasj) [reply pom.xml]: yup, that's better. Thanks

**Review by lukasj — APPROVED** [review]

**Pull request merged**